assignment/assignment.hs

```diff
@@ -42,8 +46,34 @@ import Data.Char(intToDigit)
 -- out how Haskell worked this out.)
   
 -- myand:: [ Bool ] -> Bool 
+count [] = 0 
+count (x:xs) =
+    n + count xs
+    where                           
+        n = if x then 1 else 0
+        
+myand a = 
+  if count a == length a     -- I don't think I have library functions enabled/working
+    then True
+    else False
+
+myand_extra a =
+  if length (myfilter (>0) a) == length a
+    then True
+    else False
+
+--alternative way if fromEnum worked
+--myand a = 
+--  if fromEnum(foldl (+) 0 a) == length a     -- whoops, I don't think I have library functions enabled/working
+--   then True
+--   else False
 
 -- myor:: [ Bool ] -> Bool 
+myor a = 
+  if count a > 0     -- whoops, I don't think I have library functions enabled/working
+    then True
+    else False
+
 
```

Comment:
You're going to hate me:

`foldl` accumulates a result by applying a function to a partial result and the next value, so

~~~ hs
myand = foldl (&&) True

myor:: [ Bool ] -> Bool 
myor = foldl (||) False
~~~